test/tools/router_check/router.cc

```diff
@@ -498,14 +524,23 @@ bool RouterCheckTool::compareResponseHeaderFields(
 }
 
 template <typename HeaderMap>
-bool RouterCheckTool::matchHeaderField(const HeaderMap& header_map,
-                                       const envoy::config::route::v3::HeaderMatcher& header,
-                                       const std::string test_type) {
+bool RouterCheckTool::matchHeaderField(
+    const HeaderMap& header_map, const envoy::config::route::v3::HeaderMatcher& header,
+    const std::string test_type,
+    envoy::RouterCheckToolSchema::HeaderMatchFailure& header_match_failure) {
   Envoy::Http::HeaderUtility::HeaderData expected_header_data{header};
   if (Envoy::Http::HeaderUtility::matchHeaders(header_map, expected_header_data)) {
     return true;
   }
 
+  header_match_failure.mutable_header_matcher()->CopyFrom(header);
+  const Http::LowerCaseString header_key(header.name());
```

Comment:
You should be able to use `getByKey()` on the header_map object

Reply:
`getByKey` is currently available for `RequestHeaderMap`s only as `ResponseHeaderMap` doesn't inherit from a `TraceContext` where the method is defined. I wouldn't want to change the `RequestHeaderMap` inheritance for now, but I've changed this header value lookup code, PTAL if it's better now